src/docquery/document.py

```diff
@@ -167,38 +176,70 @@ def preview(self) -> "Image":
         return [self.b.convert("RGB")]
 
     @cached_property
-    def context(self) -> Tuple[(str, List[int])]:
-        words, boxes = apply_tesseract(self.b, lang=None, tesseract_config="")
+    def context(self) -> Dict[str, List[Tuple["Image.Image", List[Any]]]]:
+        words, boxes = self._normalize_boxes(*self.ocr_reader.apply_ocr(self.b), self.b.width, self.b.height)
+
         return {
             "image": [
                 (
                     self.b,
-                    [x for x in zip(words, boxes)],
+                    self._make_word_boxes(words, boxes),
                 )
             ]
         }
 
 
 @validate_arguments
-def load_document(fpath: str):
+def load_document(fpath: str, ocr_reader_name=None):
     if fpath.startswith("http://") or fpath.startswith("https://"):
         resp = requests.get(fpath, stream=True)
         if not resp.ok:
             raise UnsupportedDocument(f"Failed to download: {resp.content}")
         b = resp.raw
     else:
         b = open(fpath, "rb")
-    return load_bytes(b, fpath)
+    return load_bytes(b, fpath, ocr_reader_name=ocr_reader_name)
 
 
-def load_bytes(b, fpath):
+def load_bytes(b, fpath, ocr_reader_name: Optional[str]):
+    ocr_reader = get_ocr_reader(ocr_reader_name)
     extension = os.path.basename(fpath).rsplit(".", 1)[-1].split("?")[0].strip()
     if extension in ("pdf"):
-        return PDFDocument(b.read())
+        return PDFDocument(b.read(), ocr_reader=ocr_reader)
     else:
         use_pil()
         try:
             img = Image.open(b)
         except UnidentifiedImageError as e:
             raise UnsupportedDocument(e)
-        return ImageDocument(img)
+        return ImageDocument(img, ocr_reader=ocr_reader)
+
+
+tesseract_reader = None
+easy_ocr_reader = None
+
+
+def get_ocr_reader(ocr_reader_name: Optional[str]):
+    global tesseract_reader
+    global easy_ocr_reader
+    if not ocr_reader_name:
+        if TESSERACT_AVAILABLE:
+            if not tesseract_reader:
+                tesseract_reader = TesseractReader()
+            return tesseract_reader
+        elif EASYOCR_AVAILABLE:
+            if not easy_ocr_reader:
+                easy_ocr_reader = EasyOCRReader()
+            return easy_ocr_reader
+        else:
+            return DummyOCRReader()
+    elif ocr_reader_name.lower() == "easyocr" and EASYOCR_AVAILABLE:
+        if not easy_ocr_reader:
+            easy_ocr_reader = EasyOCRReader()
```

Comment:
I also mistakenly instantiated `EasyOcrReader()` directly while trying to use this. I think to make this clearer, we should (a) move this and the global state into the `ocr_reader` package and (b) add a param in `OcrReader.__init__` that's `singleton=False`, pass `singleton=True`, and assert that `singleton == True`, so that users don't accidentally construct them directly.